src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerInitializationHostService.cs

```diff
@@ -55,6 +55,15 @@ private async Task ApplyStartContextIfPresent()
 
                 bool success = _instanceManager.StartAssignment(assignmentContext);
                 _logger.LogInformation($"StartAssignment invoked (Success={success})");
+
+                if (assignmentContext.MSIContext != null)
```

Comment:
Why is this logic different for V2? The V3 PR doesn't null check this I see.

Reply:
Yes :)